trie/trie.go

```diff
@@ -73,17 +73,34 @@ type (
 // NewTrie creates a trie with DB filename
 func NewTrie(kvStore db.KVStore, name string, root hash.Hash32B) (Trie, error) {
 	if kvStore == nil {
-		return nil, errors.New("try to create trie with empty KV store")
+		return nil, errors.Wrapf(ErrInvalidTrie, "try to create trie with empty KV store")
 	}
-	return newTrie(kvStore, name, root), nil
+	t := &trie{
+		cb:        db.NewCachedBatch(),
+		dao:       kvStore,
+		rootHash:  root,
+		bucket:    name,
+		numEntry:  1,
+		numBranch: 1,
+	}
+	t.lifecycle.Add(kvStore)
+	return t, nil
 }
 
 // NewTrieSharedBatch creates a trie with a shared batch
 func NewTrieSharedBatch(kvStore db.KVStore, batch db.CachedBatch, name string, root hash.Hash32B) (Trie, error) {
 	if kvStore == nil || batch == nil {
-		return nil, errors.New("try to create trie with empty KV store")
+		return nil, errors.Wrapf(ErrInvalidTrie, "try to create trie with empty KV store")
 	}
-	return newTrieSharedBatch(kvStore, batch, name, root), nil
+	t := &trie{
+		cb:        batch,
+		dao:       kvStore,
+		rootHash:  root,
+		bucket:    name,
+		numEntry:  1,
+		numBranch: 1}
+	t.lifecycle.Add(kvStore)
```

Comment:
I doubt this is a safe operation:

1. We don't want to start/stop kv store upon each batch/block cycle
2. We don't know kvStore is started/stopped here. It's better for the creator to start/stop it.

Reply:
In fact, I think it's never been used, because working set doesn't use lifecycle

Reply:
yes, this can be removed, since the kvStore here must have been already added